User account page - theming (issue/8201)

kingfisher88 opened the pull request.

kingfisher88 committed:
User account page theming

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8201

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8201

dzadza commented on web/themes/custom/recipe_theme/src/sass/block/_block.scss:
```diff
@@ -0,0 +1,12 @@
+  h2 {
+    display: flex;
+    justify-content: center;
+    margin: 2rem auto 1rem auto;
```
this can be simplified to `margin: 2rem auto 1rem;
https://www.w3schools.com/css/css_margin.asp

dzadza commented on web/themes/custom/recipe_theme/src/sass/block/_block.scss:
```diff
@@ -0,0 +1,12 @@
+    display: flex;
+    justify-content: center;
+    margin: 2rem auto 1rem auto;
+    font-family: Dosis;
```
Is this font defined somewhere?

dzadza commented on web/themes/custom/recipe_theme/src/sass/block/_block.scss:
```diff
@@ -0,0 +1,12 @@
+.layout__region {
+  h2 {
```
These rules apply to all the `h2` headings in `layout__region`. Is this intentional?

dzadza replied:
> These rules apply to all the `h2` headings in `layout__region`. Is this intentional?

dzadza commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,42 @@
+  .field__item {
+    text-align: justify;
+    text-indent: 2rem;
+    font-family: Open Sans;
```
I think this needs to be in quotes.

dzadza commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,42 @@
+    font-weight: normal;
+    font-size: 1rem;
+    line-height: 1.5rem;
+    color: #273238;
```
This hexa code is used is several places. It's worth creating a variable in this case. It's less error prone and easier to override if ever necessary.

dzadza commented on web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list.html.twig:
```diff
@@ -0,0 +1,32 @@
+		<div{{row.attributes.addClass(row_classes)}}>
+			{{- row.content -}}
+		</div>
+	{% endfor %}
```
Fix the indentation please.

kingfisher88 committed:
Simplify the margin settings
Define the Dosis font
Change the .layout_region class to your_recipes class
Use variables for color instead of hexa code
Fix indentation

dzadza commented on web/themes/custom/recipe_theme/recipe_theme.info.yml:
```diff
@@ -20,6 +20,7 @@ regions:
 libraries:
   - radix/style
   - recipe_theme/style
+  - 'recipe_theme/fonts'
```
why the quotes?

dzadza commented on web/themes/custom/recipe_theme/recipe_theme.libraries.yml:
```diff
@@ -8,3 +8,8 @@ style:
+fonts:
+  css:
+    theme:
+      '//fonts.googleapis.com/css2?family=Dosis&display=swap': { external: true }
```
Please download the fonts to your codebase. So it won't depend on an external source.

dzadza commented on web/themes/custom/recipe_theme/src/components/block/block--views-block--recipe-list-block-1.html.twig:
```diff
@@ -0,0 +1,36 @@
+
+	{% if html_tag %}
+	</{{html_tag}}>
+{% endif %}
```
The indentation is still off.

dzadza commented on web/themes/custom/recipe_theme/src/components/block/block--views-block--recipe-list-block-1.html.twig:
```diff
@@ -0,0 +1,36 @@
+{#
```
As this file is a component, you can omit the `.html`

kingfisher88 committed:
Add inernal custom fonts

kingfisher88 committed:
Save changes for chenging the branch

kingfisher88 committed:
Save work

kingfisher88 committed:
Add 'Dosis' font-family
Theme the "Your recipe" part of acount page according to wireframe

kingfisher88 committed:
Config export

gkurucz commented on web/themes/custom/recipe_theme/src/components/block/block--views-block--recipe-list-block-1.twig:
```diff
@@ -0,0 +1,36 @@
+{#
```
If you'd like to alter the template of a view please use view templates for that. Block is just the 'embedding wrapper' of any element in Drupal.

gkurucz commented on web/themes/custom/recipe_theme/src/sass/base/_font_faces.scss:
```diff
@@ -0,0 +1,41 @@
+@font-face {
+  font-family: 'Dosis';
+  src: url(../../fonts/dosis/Dosis-ExtraLight.woff);
```
Please us the [practical level of browser support](https://css-tricks.com/snippets/css/using-font-face/#article-header-id-1). Also add `woff2` formats for the fonts.

gkurucz commented on web/themes/custom/recipe_theme/src/sass/block/_block.scss:
```diff
@@ -0,0 +1,18 @@
+.your_recipes {
```
We aim to use BEM as much as possible at Pronovix, so the only acceptable format of using underscores is when you select the element of a block in the BEM context. http://getbem.com/naming/

Also, if I'm not mistaken Balazs already figured out a way to automatically include the view's name in a twig template so you don't need to redefine a name here. Please consult him regarding this.

gkurucz commented on web/themes/custom/recipe_theme/src/sass/block/_block.scss:
```diff
@@ -0,0 +1,18 @@
+  }
+}
+
+.your_recipe_content {
```
Same as my previous comment regarding naming.

gkurucz commented on web/themes/custom/recipe_theme/templates/block/block--views-block--recipe-list-block-1.html.twig:
```diff
@@ -0,0 +1,9 @@
+{#
```
Same as my comment at https://github.com/Pronovix/food/pull/45/files#diff-3219f52700428c5eb5ccfc3a175a396aR1

gkurucz commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,59 @@
+
+.user__name,
+.user__field-first-name {
+  .field__label {
```
You don't need to move this nested rule in a separate block. You could add this right after line 13. Same goes for other similar rules.

gkurucz commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,59 @@
+    display: flex;
+    justify-content: center;
+    font-family: 'Dosis', sans-serif;
+    font-weight: 300;
```
If you overwrite this in `field__label` why not just apply this rule to `field__item`?

gkurucz commented on web/themes/custom/recipe_theme/src/sass/block/_block.scss:
```diff
@@ -0,0 +1,18 @@
+.your_recipes {
+  h2 {
+    display: flex;
+    justify-content: center;
```
If the sole purpose here is to align this element to the center then `text-align: center;` is enough. Flex is overkill just for center aligning text.

gkurucz commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,59 @@
+  .field__label,
+  .field__item {
+    display: flex;
+    justify-content: center;
```
Same as my previous comment regarding center alignment.

kingfisher88 committed:
Add woff2  format to Dosis fonts
Rename classes according to BEM
Simplify the rules for some classes, remove the duplicated rules

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8201

gkurucz commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,55 @@
+}
+
+.user__field-introduction {
+  .field__label {
```
This can be still further simplified. There are rules here which are the same as 

```sass
.user__name,
.user__field-first-name {
  .field__label {
    rules...
  }
}
```

Please combine the common rules.

gkurucz commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,55 @@
+    margin-top: 1.5rem;
+    font-family: 'Dosis', sans-serif;
+    font-style: normal;
+    font-weight: normal !important;
```
Why is this `!important` needed?

gkurucz commented on web/themes/custom/recipe_theme/templates/views/views-view-unformatted--catalog.html.twig:
```diff
@@ -16,18 +16,19 @@
  */
```
Fix indentation.

gkurucz replied:
Fix indentation.

kingfisher88 committed:
Fix indentation
Remove unnecessery scc rules

gkurucz commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,47 @@
+}
+
+.user__field-introduction {
+  .field__label {
```
This can be still further simplified. There are rules here which are the same as 

```sass
.user__name,
.user__field-first-name {
  .field__label {
    rules...
  }
}
```

Please combine the common rules.

gkurucz commented on web/themes/custom/recipe_theme/src/components/block/block--views-block--recipe-list-block-1.twig:
```diff
@@ -0,0 +1,36 @@
+{#
+/**
```
Please also fix the indentation here. (indentation = 2 spaces)

kingfisher88 committed:
Simplify scc rules.
Fix indentation.

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8201

gkurucz commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,53 @@
+}
+
+.user__user-picture {
+  display: flex;
```
If the only purpose here is to center align an image then `text-align: center;` should be enough. Flex is overkill just for horizontally center aligning stuff.

kingfisher88 committed:
Fix indentation.
Change css rule to text-align: center for user picture.

gkurucz commented on web/themes/custom/recipe_theme/src/sass/view/_user_account_page.scss:
```diff
@@ -0,0 +1,52 @@
+  .field__item {
+    text-align: justify;
+    text-indent: 2rem;
+    font-family: 'Open Sans';
```
This `font-family` should also have a fallback like https://github.com/Pronovix/food/pull/45/files#diff-b4ba1b654245460d8de95a140b9e8effR6

kingfisher88 committed:
Add fallback font-family to user introduction

gkurucz commented on config/sync/usergreeting.settings.yml:
```diff
@@ -0,0 +1,3 @@
+morning_start: 19
```
Hmm in what country does the morning start at 19? :thinking:

gkurucz commented on config/sync/block.block.recipe_theme_page_title.yml:
```diff
@@ -17,4 +19,9 @@ settings:
+visibility:
+  request_path:
+    id: request_path
+    pages: '/user/*'
```
If the Page Title is disabled on the user page, will that page have a `h1` element? From a SEO and Accessibility perspective it's important that a webpage has at least and preferably one `h1` element.

gkurucz replied:
I think this question still wasn't addressed.

kingfisher88 committed:
Implements hook_preprocess_HOOK() for page.html.twig
Hide page title from user page

kingfisher88 committed:
Config export

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8201

kingfisher88 committed:
Revert "Config export"

This reverts commit f073676b124c1d342e88415207ba43e4bacc6a84.

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8201

kingfisher88 committed:
Add title to the contents in recipe_theme
Config export:
- change Content view mode to Teaser
- change Content types-Article- Manage display

gkurucz commented on web/themes/custom/recipe_theme/templates/page/page.html.twig:
```diff
@@ -0,0 +1,67 @@
+#}
+{% set page_classes = [
+  'page',
+  is_userpage ? 'page--user',
```
Although the solution you provided in `page.inc` can be useful in many cases, core's `classy` theme provides a more general and reusable solution that is not specific for user pages. Please see `web/core/themes/classy/templates/layout/html.html.twig` line `29`.

gkurucz commented on web/themes/custom/recipe_theme/includes/page.inc:
```diff
@@ -0,0 +1,22 @@
+  $variables['page_attributes'] = new Attribute();
+  $route_name = \Drupal::routeMatch()->getRouteName();
+  if ($route_name === 'entity.user.canonical') {
+    $variables['is_userpage'] = TRUE;
```
This is a nice solution, but a little overkill. Please see my other comment at https://github.com/Pronovix/food/pull/45/files#diff-b2f684ce1ef5fd62d13500f925ecb33bR9

kingfisher88 committed:
Set classes to pages according to their root_path

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8201

 Conflicts:
	config/sync/core.entity_view_display.node.article.teaser.yml
	config/sync/usergreeting.settings.yml
	config/sync/views.view.recipe_list.yml
	web/themes/custom/recipe_theme/src/sass/block/_block.scss

kingfisher88 committed:
Fix conflocts

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8201
